src/index.re

```diff
@@ -94,25 +99,26 @@ let getDirection = ({direction, pos: (x, y), nextDirection}) => {
   };
 };
 
-let move = ({direction, pos: (x, y), nextDirection}) => {
+let move = (agentState: agent) => {
+  let (prevX, prevY) = agentState.pos;
   let x =
-    switch (direction) {
-    | Right => x == GridSizes.gridWidthf ? 0. : x
-    | Left => x == 0. ? GridSizes.gridWidthf : x
-    | _ => x
+    switch (agentState.direction) {
+    | Right => prevX == GridSizes.gridWidthf ? 0. : prevX
+    | Left => prevX == 0. ? GridSizes.gridWidthf : prevX
+    | _ => prevX
     };
   let y =
-    switch (direction) {
-    | Up => y == 0. ? GridSizes.gridHeightf : y
-    | Down => y == GridSizes.gridHeightf ? 0. : y
-    | _ => y
+    switch (agentState.direction) {
+    | Up => prevY == 0. ? GridSizes.gridHeightf : prevY
+    | Down => prevY == GridSizes.gridHeightf ? 0. : prevY
+    | _ => prevY
     };
-  let direction = getDirection({direction, nextDirection, pos: (x, y)});
+  let direction = getDirection({...agentState, pos: (x, y)});
   switch (direction) {
-  | Up => {direction, nextDirection, pos: (x, y -. 1.)}
-  | Down => {direction, nextDirection, pos: (x, y +. 1.)}
-  | Right => {direction, nextDirection, pos: (x +. 1., y)}
-  | Left => {direction, nextDirection, pos: (x -. 1., y)}
+  | Up => {...agentState, direction, pos: (x, y -. agentState.velocity)}
+  | Down => {...agentState, direction, pos: (x, y +. agentState.velocity)}
+  | Right => {...agentState, direction, pos: (x +. agentState.velocity, y)}
+  | Left => {...agentState, direction, pos: (x -. agentState.velocity, y)}
```

Comment:
This part is just unrelated small improvements:
* spread state
* improve variable names